packages/mwp-app-server/src/util/getPlugins.js

```diff
@@ -20,23 +20,41 @@ import { plugin as requestAuthPlugin } from 'mwp-auth-plugin';
 const CSRF_COOKIE_NAME = 'x-mwp-csrf';
 const CSRF_HEADER_COOKIE_NAME = `${CSRF_COOKIE_NAME}-header`;
 const CSRF_HEADER_NAME = CSRF_COOKIE_NAME;
+// dev cookie names
+const DEV_CSRF_COOKIE_NAME = `${CSRF_COOKIE_NAME}-dev`;
+const DEV_CSRF_HEADER_COOKIE_NAME = `${DEV_CSRF_COOKIE_NAME}-header`;
+const DEV_CSRF_HEADER_NAME = DEV_CSRF_COOKIE_NAME;
 
+// Sets the csrf header token from plugin into a cookie
 export function setCsrfCookies(request, h) {
-	const csrfHeader = (request.response.headers || {})[CSRF_COOKIE_NAME];
+	const { headers = {} } = request.response;
+	const csrfHeader = headers[CSRF_COOKIE_NAME];
+	const csrfDevHeader = headers[DEV_CSRF_COOKIE_NAME];
 	if (csrfHeader) {
 		h.state(CSRF_HEADER_COOKIE_NAME, csrfHeader);
 	}
+	if (csrfDevHeader) {
+		h.state(DEV_CSRF_HEADER_COOKIE_NAME, csrfDevHeader);
+	}
 	return h.continue;
 }
 
 /**
- * The CSRF plugin we use - 'electrode-csrf-jwt' compares a cookie token to a
- * header token in non-GET requests. By default, it will set the cookie token
- * itself (CSRF_COOKIE_NAME), and supply the corresponding header token in a custom
- * header (also CSRF_COOKIE_NAME). However, we update this flow to also supply the
- * header token as a cookie (CSRF_COOKIE_NAME-header) so that it syncs across
- * browser tabs.
+ * The CSRF plugin we use, 'electrode-csrf-jwt', generates a token for each request 
+ * that we make and sets the token in an HTTP-only cookie (CSRF_COOKIE_NAME) and in
+ * the HTTP response header (also CSRF_COOKIE_NAME). In non-GET requests we must supply
```

Comment:
I don't think the plugin makes any determination of which request method it will apply to (GET vs other) - it's up to the routes to opt in/out of CSRF protection.

Reply:
It actually does check the request method - https://github.com/meetup/electrode-csrf-jwt/blob/master/lib/csrf.js#L60
Or you can specify a 'skipVerify' function as an option, but the default behavior when using this plugin is to protect POST/PATCH/PUT/DELETE requests